### PR TITLE
Add io.github.unknownskl.greenlight

### PR DESCRIPTION
**Pull request by unknownskl**
### Please confirm your submission meets all the criteria

- [x] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [x] My pull request follows the instructions at [App Submission][submission].
- [x] I have [built][build] and tested the submission locally.
- [x] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
    - `--device=all` is needed for controller access
    - `- --filesystem=/run/udev:ro` seemed to be required to read the controllers properly (Please correct me if im wrong about this one)
- [x] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [x] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link:**
- [x] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [x] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

[reqs]: https://docs.flathub.org/docs/for-app-authors/requirements
[maint]: https://docs.flathub.org/docs/for-app-authors/maintanance
[submission]: https://docs.flathub.org/docs/for-app-authors/submission
[build]: https://docs.flathub.org/docs/for-app-authors/submission/#before-submission

### CODE REVIEW

File: io.github.unknownskl.greenlight.yml [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+      - yarn run flatpak-build --offline -- $ELECTRON_BUILDER_ARCH_ARGS --linux
+      # Add icon, metadata and desktop file
+      - install -Dm 644 /run/build/greenlight/io.github.unknownskl.greenlight.png ${FLATPAK_DEST}/share/icons/hicolor/512x512/apps/${FLATPAK_ID}.png
+      - install -Dm 644 /run/build/greenlight/io.github.unknownskl.greenlight.desktop ${FLATPAK_DEST}/share/applications/${FLATPAK_ID}.desktop
+      - install -Dm 644 /run/build/greenlight/io.github.unknownskl.greenlight.metainfo.xml ${FLATPAK_DEST}/share/metainfo/${FLATPAK_ID}.appdata.xml
+      # Bundle app and dependencies
+      - cp -a dist/linux*unpacked /app/main
+      # Install app wrapper
+      - install -Dm755 -t /app/bin/ ../run.sh
+      - ls -la /app /app/bin /app/main
```
Comment (hfiguiere) [inline io.github.unknownskl.greenlight.yml:46]:
```suggestion
```

File: io.github.unknownskl.greenlight.yml [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+        npm_config_offline: 'true'
+        yarn_config_cache: /run/build/greenlight/flatpak-node/yarn-mirror
+    build-commands:
+      - cp -a ../extralib -t /app/
+      - yarn --offline
+      - yarn run flatpak-build --offline -- $ELECTRON_BUILDER_ARCH_ARGS --linux
+      # Add icon, metadata and desktop file
+      - install -Dm 644 /run/build/greenlight/io.github.unknownskl.greenlight.png ${FLATPAK_DEST}/share/icons/hicolor/512x512/apps/${FLATPAK_ID}.png
+      - install -Dm 644 /run/build/greenlight/io.github.unknownskl.greenlight.desktop ${FLATPAK_DEST}/share/applications/${FLATPAK_ID}.desktop
+      - install -Dm 644 /run/build/greenlight/io.github.unknownskl.greenlight.metainfo.xml ${FLATPAK_DEST}/share/metainfo/${FLATPAK_ID}.appdata.xml
```
Comment (hfiguiere) [inline io.github.unknownskl.greenlight.yml:41]:
```suggestion
      - install -Dm 644 ../io.github.unknownskl.greenlight.png ${FLATPAK_DEST}/share/icons/hicolor/512x512/apps/${FLATPAK_ID}.png
      - install -Dm 644 ../io.github.unknownskl.greenlight.desktop ${FLATPAK_DEST}/share/applications/${FLATPAK_ID}.desktop
      - install -Dm 644 ../io.github.unknownskl.greenlight.metainfo.xml ${FLATPAK_DEST}/share/metainfo/${FLATPAK_ID}.appdata.xml
```

File: icon_512.png
Comment (hfiguiere) [inline icon_512.png:1]: this file is never referenced.

File: io.github.unknownskl.greenlight.metainfo.xml [hunk 3]
```diff
@@ -0,0 +1,29 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<component type="desktop">
+  <id>io.github.unknownskl.greenlight</id>
+  <launchable type="desktop-id">io.github.unknownskl.greenlight.desktop</launchable>
+  <name>Greenlight</name>
+  <project_license>MIT</project_license>
+  <developer_name>UnknownSKL</developer_name>
+  <summary>Greenlight is an open-source client for xCloud and Xbox home streaming.</summary>
```
Comment (hfiguiere) [inline io.github.unknownskl.greenlight.metainfo.xml:8]:
```suggestion
  <summary>Greenlight is an open-source client for xCloud and Xbox home streaming</summary>
```

File: flathub.json [hunk 4]
```diff
@@ -0,0 +1,3 @@
+{
+    "only-arches": ["x86_64"]
```
Comment (hfiguiere) [inline flathub.json:2]: why?

File: io.github.unknownskl.greenlight.yml [hunk 5]
```diff
@@ -0,0 +1,79 @@
+app-id: io.github.unknownskl.greenlight
+runtime: org.freedesktop.Platform
+runtime-version: '23.08'
+sdk: org.freedesktop.Sdk
+base: org.electronjs.Electron2.BaseApp
+base-version: '23.08'
+sdk-extensions:
+  - org.freedesktop.Sdk.Extension.node18
+platform-extensions:
+  - org.freedesktop.Platform.VAAPI.Intel
```
Comment (hfiguiere) [inline io.github.unknownskl.greenlight.yml:10]:
```suggestion
```

This is not a runtime.

File: flathub.json (hunk 4 above)
Comment (unknownskl) [reply flathub.json:2]: Not sure actually, couldn't test it myself. The project should in theory work on aarch64, will remove the flathub.json file. Perhaps i can always limit it later when users are reporting it not to be working?

File: io.github.unknownskl.greenlight.yml [hunk 6]
```diff
@@ -0,0 +1,79 @@
...
+    build-options:
+      env:
+        XDG_CACHE_HOME: /run/build/greenlight/flatpak-node/cache
+        npm_config_cache: /run/build/greenlight/flatpak-node/npm-cache
+        npm_config_offline: 'true'
+        yarn_config_cache: /run/build/greenlight/flatpak-node/yarn-mirror
+    build-commands:
+      - cp -a ../extralib -t /app/
+      - yarn --offline
+      - yarn run flatpak-build --offline -- $ELECTRON_BUILDER_ARCH_ARGS --linux
```
Comment (hfiguiere) [inline io.github.unknownskl.greenlight.yml:37]: where is `ELECTRON_BUILDER_ARCH_ARGS` defined?

File: io.github.unknownskl.greenlight.yml (hunk 5 above)
Comment (unknownskl) [reply io.github.unknownskl.greenlight.yml:10]: Thanks, first time building a flatpak. Will remove it

File: io.github.unknownskl.greenlight.yml (hunk 6 above)
Comment (unknownskl) [reply io.github.unknownskl.greenlight.yml:37]: It is a leftover from an example flatpak file. Will remove it, thanks.

File: icon_512.png
Comment (unknownskl) [reply icon_512.png:1]: File has been removed

**Comment by hfiguiere** [comment]
bot, build io.github.unknownskl.greenlight

**Review by bbhtt — COMMENTED** [review]
Can you submit the desktop file and appdata upstream?

File: io.github.unknownskl.greenlight.yml [hunk 7]
```diff
@@ -0,0 +1,76 @@
...
+      - type: file
+        path: io.github.unknownskl.greenlight.png
+      - type: file
+        path: io.github.unknownskl.greenlight.desktop
+      # Wrapper to launch the app
+      - type: script
+        dest-filename: run.sh
+        commands:
+          - export LIBGL_DRIVERS_PATH=/app/extralib/dri:/usr/lib/x86_64-linux-gnu/dri/intel-vaapi-driver:/usr/lib/x86_64-linux-gnu/dri:${LIBGL_DRIVERS_PATH}
+          - export LIBVA_DRIVERS_PATH=/app/extralib/dri:/usr/lib/x86_64-linux-gnu/dri/intel-vaapi-driver:/usr/lib/x86_64-linux-gnu/dri:${LIBVA_DRIVERS_PATH}
```
Comment (bbhtt) [inline io.github.unknownskl.greenlight.yml:72]:
```
3:1214/083616.905801:ERROR:bus.cc(399)] Failed to connect to the bus: Failed to connect to socket /run/dbus/system_bus_socket: No such file or directory
MESA-LOADER: failed to open radeonsi: /usr/lib/x86_64-linux-gnu/dri/radeonsi_dri.so: cannot open shared object file: No such file or directory (search paths /app/extralib/dri:/usr/lib/x86_64-linux-gnu/dri/intel-vaapi-driver:/usr/lib/x86_64-linux-gnu/dri:, suffix _dri)
failed to load driver: radeonsi
MESA-LOADER: failed to open swrast: /usr/lib/x86_64-linux-gnu/dri/swrast_dri.so: cannot open shared object file: No such file or directory (search paths /app/extralib/dri:/usr/lib/x86_64-linux-gnu/dri/intel-vaapi-driver:/usr/lib/x86_64-linux-gnu/dri:, suffix _dri)
failed to load driver: swrast
```

There is nothing in there:

```
flatpak run --command=sh io.github.unknownskl.greenlight
[📦 io.github.unknownskl.greenlight ~]$ ls /usr/lib/x86_64-linux-gnu/dri/
intel-vaapi-driver
```

And this are not set in the runtime.

Comment (bbhtt) [reply io.github.unknownskl.greenlight.yml:72]: The extension point is GL:

```
[📦 io.github.unknownskl.greenlight ~]$ ls /usr/lib/x86_64-linux-gnu/GL/default/lib/dri/
crocus_dri.so  iris_dri.so	  libgallium_dri.so	   nouveau_dri.so	 r300_dri.so  r600_drv_video.so  radeonsi_drv_video.so	virtio_gpu_dri.so	 vmwgfx_dri.so
i915_dri.so    kms_swrast_dri.so  libgallium_drv_video.so  nouveau_drv_video.so  r600_dri.so  radeonsi_dri.so	 swrast_dri.so		virtio_gpu_drv_video.so  zink_dri.so
[📦 io.github.unknownskl.greenlight ~]$ 
```

Comment (bbhtt) [reply io.github.unknownskl.greenlight.yml:72]: Also you're hardcoding the path to x86-64, when the flatpak is built for aarch64 too: `$(gcc --print-multiarch)`

Comment (hfiguiere) [reply io.github.unknownskl.greenlight.yml:72]: but I doubt intel-vaapi exist on that platform.

**Comment by unknownskl** [comment]
> Can you submit the desktop file and appdata upstream?

What do you mean? The desktop file and the appdata are in both the repository and the flatpak build. Do you want me to put it somewhere else?

File: io.github.unknownskl.greenlight.yml (hunk 7 above)
Comment (unknownskl) [reply io.github.unknownskl.greenlight.yml:72]: Thanks, valid point. Will make some changes and let you know once it has been solved

**Comment by bbhtt** [comment]
> > Can you submit the desktop file and appdata upstream?
> 
> What do you mean? The desktop file and the appdata are in both the repository and the flatpak build. Do you want me to put it somewhere else?

Upstream is here https://github.com/unknownskl/greenlight, screenshots should come from them too.

**Comment by hfiguiere** [comment]
> The desktop file and the appdata are in both the repository and the flatpak build.

if they are in the upstream repository then they should be fetch from there and not be included here.

**Comment by hfiguiere** [comment]
same for the icon.

File: io.github.unknownskl.greenlight.yml (hunk 7 above)
Comment (unknownskl) [reply io.github.unknownskl.greenlight.yml]: Got the error reproduced on my Steam deck and the new commit should fix the error

**Comment by unknownskl** [comment]
Thanks, i've moved the assets and such to the upstream repo: https://github.com/unknownskl/greenlight/tree/main-v2/flatpak

**Comment by unknownskl** [comment]
bot, build io.github.unknownskl.greenlight

File: io.github.unknownskl.greenlight.yml (hunk 7 above)
Comment (bbhtt) [reply io.github.unknownskl.greenlight.yml]: You forgot to unhardcode the x86_64 paths, the flatpak is built for aarch64 too.

**Pull request closed without merging**